posthog/schema_helpers.py

```diff
@@ -1,81 +1,14 @@
+import orjson
 from pydantic import BaseModel
+from rest_framework.utils.encoders import JSONEncoder
 
-from posthog.schema import (
-    FunnelStepReference,
-    FunnelsFilter,
-    FunnelsQuery,
-    FunnelVizType,
-    FunnelConversionWindowTimeUnit,
-    FunnelLayout,
-    BreakdownAttributionType,
-    StepOrderValue,
-)
 
+def to_json(query: BaseModel) -> bytes:
+    instance_dict = query.model_construct(**query.model_dump(exclude_none=True, exclude_defaults=False)).model_dump(
```

Comment:
Sorry, not a helpful comment, but this looks like a lot of dumping around just to get to a cache key in the end 🙈

Reply:
Sorry, that's already outdated. See new code for a separate approach (manually re-adding the literal defaults). I'm open for other suggestions though - haven't figured out a clean solution that doesn't require us to write the models backend side yet.